### PR TITLE
Fix sprintf call in AdminCustomers

### PR DESCRIPTION
**Pull request by prestamodule**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | As now, the %s is replaced with "htmlspecialchars"
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | Check the Pull Request and edit a customer.

### CODE REVIEW

**Comment by xBorderie** [comment]
Thanks! It seems to also happen in the Theme page :/
http://forge.prestashop.com/browse/BOOM-1896
ping @AlexEven

File: controllers/admin/AdminCustomersController.php [hunk 1]
```diff
@@ -249,15 +249,15 @@ public function initToolbarTitle()
                 /** @var Customer $customer */
                 if (($customer = $this->loadObject(true)) && Validate::isLoadedObject($customer)) {
                     array_pop($this->toolbar_title);
-                    $this->toolbar_title[] = sprintf($this->trans('Information about customer %s', array(), 'Admin.OrdersCustomers.Feature'), Tools::substr($customer->firstname, 0, 1).'. '.$customer->lastname);
+                    $this->toolbar_title[] = sprintf($this->trans('Information about customer %s', array('%s' => Tools::substr($customer->firstname, 0, 1).'. '.$customer->lastname), 'Admin.OrdersCustomers.Feature'));
```
Comment (antoin-m) [inline controllers/admin/AdminCustomersController.php]: You may want to remove the call to `sprintf` if there is no need for it.

File: controllers/admin/AdminCustomersController.php [hunk 2]
```diff
@@ -249,15 +249,15 @@ public function initToolbarTitle()
...
                 }
                 break;
             case 'add':
             case 'edit':
                 array_pop($this->toolbar_title);
                 /** @var Customer $customer */
                 if (($customer = $this->loadObject(true)) && Validate::isLoadedObject($customer)) {
-                    $this->toolbar_title[] = sprintf($this->trans('Editing customer %s', array(), 'Admin.OrdersCustomers.Feature'), Tools::substr($customer->firstname, 0, 1).'. '.$customer->lastname);
+                    $this->toolbar_title[] = sprintf($this->trans('Editing customer %s.', array('%s' => Tools::substr($customer->firstname, 0, 1).'. '.$customer->lastname), 'Admin.OrdersCustomers.Feature'));
```
Comment (antoin-m) [inline controllers/admin/AdminCustomersController.php]: You may want to remove the call to `sprintf` if there is no need for it.

File: controllers/admin/AdminCustomersController.php [hunk 3]
```diff
@@ -397,7 +397,7 @@ public function renderForm()
                     'required' => ($obj->id ? false : true),
                     'col' => '4',
                     'hint' => ($obj->id ? $this->trans('Leave this field blank if there\'s no change.', array(), 'Admin.OrdersCustomers.Help') :
-                        sprintf($this->trans('Password should be at least %s characters long.', array(), 'Admin.OrdersCustomers.Help'), Validate::PASSWORD_LENGTH))
+                        sprintf($this->trans('Password should be at least %s characters long.', array('%s' => Validate::PASSWORD_LENGTH), 'Admin.OrdersCustomers.Help')))
```
Comment (antoin-m) [inline controllers/admin/AdminCustomersController.php]: You may want to remove the call to `sprintf` if there is no need for it.

**Review by antoin-m — CHANGES_REQUESTED** [review]

**Comment by maximebiloe** [comment]
Same as here: https://github.com/PrestaShop/PrestaShop/pull/7124

Could you test this one? https://github.com/PrestaShop/PrestaShop/pull/7087

**Comment by PrestaEdit** [comment]
Hi @maximebiloe 

It's seems not working with the PR you mention...

**Comment by maximebiloe** [comment]
Hello @prestamodule,

The PR https://github.com/PrestaShop/PrestaShop/pull/7194 solve the "htmlspecialchars" issue, so this one is no longer needed.
Thank you for your help.

Regards

**Pull request closed without merging**